src/nav/navActions.js

```diff
@@ -10,10 +11,24 @@ import type {
   UserOrBot,
   ApiResponseServerSettings,
 } from '../types';
+
+import type { Action } from '../actionTypes';
 import { getSameRoutesCount } from '../selectors';
 
-export const navigateBack = () => (dispatch: Dispatch, getState: GetState): NavigationAction =>
-  dispatch(StackActions.pop({ n: getSameRoutesCount(getState()) }));
+export const navigateBack = () => (
+  dispatch: Dispatch,
+  getState: GetState,
+): NavigationAction | Action => {
+  /**
+   * If a message is currently being edited, cancel the edit. Otherwise,
+   * navigate back.
+   */
```

Comment:
The code change below is very much a hack.

Its associated comment should thus start with `HACK: `, or some similar marker of grave concern, and then continue with what the right thing to do is (or is believed to be) and why this has been done here instead of that there. Similarly, some documentation of this hack's existence should be present in ComposeBox, ActionSheet, and/or some other component responsible for the edit message state.

(Also, this is function-documentation-style comment formatting [[0]](https://jsdoc.app/about-getting-started.html) [[1]](https://developer.gnome.org/gtk-doc-manual/stable/documenting_syntax.html.en) [[2]](https://www.oracle.com/technetwork/articles/java/index-137868.html), rather than ordinary comment formatting.)